chatkit/src/main/java/com/pusher/chatkit/Result.kt

```diff
@@ -0,0 +1,34 @@
+package com.pusher.chatkit
+
+/**
+ * Code from https://github.com/michaelbull/kotlin-result
```

Comment:
Can we include import this with gradle a al https://github.com/michaelbull/kotlin-result#installation?

Reply:
Could do but that has a load more utility functions for interacting with the `Result` that we don't need.

Reply:
fair enough. I'm a little bit fuzzy on whether a link to the lisence is enough to fulfil

> provided that the above copyright notice and this permission notice appear in all copies

but I'm sure I'm splitting hairs here 😛